pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -251,6 +254,47 @@ func (s *Reconciler) Update(ctx context.Context) error {
 				}
 			}
 		}
+
+		// If the internal load balancer isn't set on a control plane machine,
+		// we should attempt to populate it else if the machine is later replaced,
+		// its replacement will not get attached to the load balancer.
+		if s.scope.Role() == actuators.ControlPlane && s.scope.MachineConfig.InternalLoadBalancer == "" {
+			for _, iface := range *vm.NetworkProfile.NetworkInterfaces {
+				// Get iface name from the ID
+				ifaceName := path.Base(*iface.ID)
+
+				lbsiface, err := s.interfaceLoadBalancersSvc.Get(ctx, &interfaceloadbalancers.Spec{
+					NicName:           ifaceName,
+					ResourceGroupName: s.scope.MachineConfig.ResourceGroup,
+				})
+				if err != nil {
+					klog.Errorf("Unable to get load balancers for interface %s: %v", ifaceName, err)
+					continue
+				}
+
+				lbs, err := decode.GetLoadBalancers(lbsiface)
+				if err != nil {
+					klog.Errorf("LoadBalancers interfaces get returned invalid load balancers interface, getting %T instead", lbsiface)
```

Comment:
this error message doesn't quite make sense to me, didn't we just try to get `lbsiface` ?

also, should this continue or something, the next clause after this will attempt to use `lbs`.

Reply:
I copied this from a similar pattern above. `lbsiface` is an `interface{}` which has an underlying type, which decode is expecting. If there's an error it's because the underlying type was wrong. It's a weird set of interfaces these Azure abstractions but 🤷 

It doesn't matter if we continue because ranging over an empty list won't do anything, but I can add a continue :)

Reply:
This should be an internal only error though, as long as we program everything correctly this should never happen at runtime

Reply:
> This should be an internal only error though, as long as we program everything correctly this should never happen at runtime

famous last words XD

thanks for the explanation though, it does sound a little confusing